The wagon you are using has not implemented getFileList() with quarkus platform 2.11.1.Final and later versions

When we include dependency on quarkus platform/plugin 2.11.1 and above into pom.xml then we receive 

[ERROR] Failed to execute goal org.codehaus.mojo:wagon-maven-plugin:2.0.2:copy (promote) on project quarkus: Error during performing repository copy: The wagon you are using has not implemented getFileList() -> [Help 1]

Detailed logs:
mvn clean initialize -Ppromote -e
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO]
[INFO] ----------------------------< test:quarkus >----------------------------
[INFO] Building quarkus 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ quarkus ---
[INFO]
[INFO] --- wagon-maven-plugin:2.0.2:copy (promote) @ quarkus ---
[INFO] Scanning remote file system: http://mavenrepo/nexus/content/repositories/pre-releases ...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.469 s
[INFO] Finished at: 2022-09-20T15:16:23+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:wagon-maven-plugin:2.0.2:copy (promote) on project quarkus: Error during performing reposit
ory copy: The wagon you are using has not implemented getFileList() -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:wagon-maven-plugin:2.0.2:copy (promote) on
project quarkus: Error during performing repository copy
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:375)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: Error during performing repository copy
    at org.codehaus.mojo.wagon.AbstractCopyMojo.execute (AbstractCopyMojo.java:58)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.UnsupportedOperationException: The wagon you are using has not implemented getFileList()
    at org.apache.maven.wagon.AbstractWagon.getFileList (AbstractWagon.java:988)
    at org.codehaus.mojo.wagon.shared.WagonDirectoryScanner.scandir (WagonDirectoryScanner.java:290)
    at org.codehaus.mojo.wagon.shared.WagonDirectoryScanner.scan (WagonDirectoryScanner.java:242)
    at org.codehaus.mojo.wagon.shared.DefaultWagonDownload.getFileList (DefaultWagonDownload.java:54)
    at org.codehaus.mojo.wagon.shared.DefaultWagonDownload.download (DefaultWagonDownload.java:63)
    at org.codehaus.mojo.wagon.shared.DefaultWagonCopy.copy (DefaultWagonCopy.java:67)
    at org.codehaus.mojo.wagon.CopyMojo.copy (CopyMojo.java:78)
    at org.codehaus.mojo.wagon.AbstractCopyMojo.execute (AbstractCopyMojo.java:54)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

mvn -v
Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
Maven home: d:\Tools\apache-maven-3.8.6
Java version: 11.0.8, vendor: Oracle Corporation, runtime: c:\Program Files\Java\jdk-11.0.8
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

Can be reporduced with this reduced pom.xml:

`<?xml version="1.0"?>`

`<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">`

	<modelVersion>4.0.0</modelVersion>
	<groupId>test</groupId>
	<artifactId>quarkus</artifactId>
	<version>1.0.0-SNAPSHOT</version>

	<properties>
		<promote.source.url>http://mavenrepo/nexus/content/repositories/pre-releases</promote.source.url>
		<promote.target.url>http://mavenrepo/nexus/content/repositories/releases</promote.target.url>
		<promote.repo.id>nexus</promote.repo.id>
		<promote.group.id>test</promote.group.id>
		<promote.artifact.id>${project.artifactId}</promote.artifact.id>
		<promote.version>${project.version}</promote.version>

	</properties>

	<dependencyManagement>
		<dependencies>
			<dependency>
				<groupId>io.quarkus.platform</groupId>
				<artifactId>quarkus-bom</artifactId>
				<version>2.11.1.Final</version>
				<type>pom</type>
				<scope>import</scope>
			</dependency>

		</dependencies>
	</dependencyManagement>
	<build>
		<plugins>

			<plugin>
				<groupId>io.quarkus.platform</groupId>
				<artifactId>quarkus-maven-plugin</artifactId>
				<version>2.11.1.Final</version>
				<extensions>true</extensions>
				<executions>
					<execution>
						<goals>
							<goal>build</goal>
							<goal>generate-code</goal>
							<goal>generate-code-tests</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
		</plugins>
	</build>


	<profiles>
		<profile>
			<id>promote</id>
			<activation>
				<activeByDefault>false</activeByDefault>
			</activation>
			<build>
				<plugins>
					<plugin>
						<groupId>org.codehaus.mojo</groupId>
						<artifactId>wagon-maven-plugin</artifactId>
						<version>2.0.2</version>
						<executions>
							<execution>
								<id>promote</id>
								<phase>initialize</phase>
								<goals>
									<goal>copy</goal>
								</goals>
								<configuration>
									<source>${promote.source.url}</source>
									<target>${promote.target.url}</target>
									<sourceId>${promote.repo.id}</sourceId>
									<targetId>${promote.repo.id}</targetId>
									<fromDir>${promote.group.id}/${promote.artifact.id}/${promote.version}</fromDir>
									<toDir>${promote.group.id}/${promote.artifact.id}/${promote.version}</toDir>
								</configuration>
							</execution>
						</executions>
					</plugin>
				</plugins>
			</build>
		</profile>
	</profiles>

</project>

When we use quarkus plaform 2.10.4 and earlier versions then it is working (of course for this specific pom "connection refused" is returned but this is ok).
Question is whether it is problem in wagon plugin or in quarkus platform, maybe someone could help us.
Thanks.

--- a/pom.xml
+++ b/pom.xml
@@ -84,7 +84,7 @@
   <properties>
     <maven.api.version>3.2.5</maven.api.version>
     <mojo.java.target>8</mojo.java.target>
-    <wagon.api.version>3.0.0</wagon.api.version>
+    <wagon.api.version>3.4.0</wagon.api.version>
 
     <!-- Test plugins -->
     <takari-plugin-testing.version>2.9.2</takari-plugin-testing.version>

